vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -38,16 +40,23 @@ def complete_order
 
       def create_order
         ActiveRecord::Base.transaction do
+          assign_account unless occupancy.account_id?
+
           create_order_and_detail
-          # TODO: This avoids an exception but results in un-purchased order on
-          # problem occupancy, which may not appear in the problem-order scope
-          if occupancy.account_id?
-            order.validate_order!
-            order.purchase!
-          end
+          order.validate_order!
+          order.purchase!
         end
       end
 
+      def assign_account
+        accounts = occupancy.user.accounts_for_product(occupancy.secure_room)
+        occupancy.update(account: accounts.first)
```

Comment:
I'm not sure if this first account will _always_ work.. if not, we could at least cycle through the `accounts_for_product` until `validate_order!` succeeds